src/webportal/src/app/job-submission/job-submission-page.jsx

```diff
@@ -298,7 +299,27 @@ export const JobSubmissionPage = ({
           })
           .catch(alert);
       }
-    } else if (params.get('op') !== 'resubmit') {
+    } else if (!isNil(window.localStorage.getItem('marketItem'))) {
+      const jobConfig = JSON.parse(localStorage.getItem('marketItem'));
+      localStorage.removeItem('marketItem');
+      const [
+        jobInfo,
+        taskRoles,
+        parameters,
+        ,
+        extras,
+      ] = getJobComponentsFromConfig(jobConfig, { vcNames });
+      jobInfo.name = generateJobName(jobInfo.name);
+      if (get(jobConfig, 'extras.submitFrom')) {
+        delete jobConfig.extras.submitFrom;
+      }
+      setJobProtocol(new JobProtocol(jobConfig));
+      setJobTaskRoles(taskRoles);
+      setParameters(parameters);
+      setJobInformation(jobInfo);
+      setExtras(extras);
+      setLoading(false);
```

Comment:
This piece of code is same with above. Can we wrap them to a function?

Reply:
I will refact this piece of code in next pr to resolve yaml modification issue. I'd rather seperate the two behaviors right now to keep the logic clear.